packages/rest/src/diagnostics.ts

```diff
@@ -63,6 +63,12 @@ const libDefinition = {
         default: paramMessage`Path contains parameter ${"param"} but wasn't found in given parameters`,
       },
     },
+    "missing-server-param": {
```

Comment:
Should also have diagnostics for

- server value with null or empty description
- server variable with a `@doc` element
- optional server variable without a default value.
- server variable with non-string type (string, string literal, union of string literals, enum of string literals)

Reply:
Apart from 1. feels like those are openapi limitation and we shouldn’t maybe limit ourself to that